### PR TITLE
fix: Card does not override specified focusMode based on event listeners

### LINKED ISSUE #31529: [Bug]: Card component - If focusMode = 'tab-only', and the Card has an onClick event handler, then `focusMode` regresses to `no-tab` behaviour
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
System:
    OS: Windows 10 10.0.22631
    CPU: (12) x64 12th Gen Intel(R) Core(TM) i7-1265U
    Memory: 14.86 GB / 31.83 GB
  Browsers:
    Edge: Chromium (123.0.2420.97)
    Internet Explorer: 11.0.22621.3527
```

### Are you reporting Accessibility issue?

yes

### Reproduction

https://stackblitz.com/edit/4g3qid?file=src%2Fexample.tsx

### Bug Description

If `focusMode = 'tab-only'`, and the Card has an event handler (especially 'onClick'), then `focusMode` regresses to `no-tab` behaviour (which is both undocumented and unexpected).
[This behaviour is introduced here](https://github.com/microsoft/fluentui/blob/fc5f6f5a47d1992c1fe44b6b0aab8965a1dc8230/packages/react-components/react-card/src/components/Card/useCard.ts#L41C27-L41C50).

## Actual Behavior
When the card is focusable, `onClick` event handlers are not allowed and adding them regresses the `focusMode` to `no-tab`.

## Expected Behavior
When the card is focused, `onClick` event handlers should be supported. As well as the `onKeyDown`/`onKeyUp` for pressing Enter which should match the `onClick` behavior.

### Logs

_No response_

### Requested priority

Normal

### Products/sites affected

Microsoft Loop

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by smhigley**
## Previous Behavior

It was impossible to have both a `focusMode=off` or `focusMode=tab-only` card with a click handler on the root. There are several reasons to do this (e.g. the card as a whole can be clicked as a larger hit target, and there's also a link or button within that does the same thing). The change in `focusMode` actually interferes with accessibility, because it makes it impossible to have a card with both a root click and enter/space handler, since a card with any `focusMode` apart from `off` will prevent enter from doing an author-defined action.

Authors already must provide a secondary method for performing any action put onto the card root, since any such actions will not be accessible to voice control, touch screen readers, etc.

## New Behavior

This updates the default `focusMode` to be `tab-only` when cards are interactive, but still allows users to override that with there own explicit `focusMode`. I chose `tab-only` instead of `no-tab` as the least heavy-handed tabster behavior, and the least different from the non-interactive default of `off`.

We could also consider having the default always be `off`, and not looking at whether or not there are event handlers attached to the root. I wasn't sure if there was another non-a11y reason to ensure cards are focusable when they have certain event handlers.

## Related Issue(s)

- Fixes #31529 and an [ADO issue](https://dev.azure.com/microsoftdesign/fluent-ui/_workitems/edit/19325).

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports | Baseline (minified/GZIP) | PR    | Change     |
| :---------------- | -----------------------: | ----: | ---------: |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/Card.All.fixture.js'>Card - All</abbr> | `101.225 kB`<br />`28.664 kB` | `101.234 kB`<br />`28.665 kB` | `9 B` <br />`1 B` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/Card.fixture.js'>Card</abbr> | `94.009 kB`<br />`26.813 kB` | `94.018 kB`<br />`26.813 kB` | `9 B` <br />|
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ReactComponents.fixture.js'>react-components: entire library</abbr> | `1.095 MB`<br />`270.363 kB` | `1.095 MB`<br />`270.362 kB` | `9 B` <br />`-1 B` |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports | Size (minified/GZIP) |
| ----------------- | -------------------: |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardFooter.fixture.js'>CardFooter</abbr> | `14.356 kB`<br />`5.798 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardHeader.fixture.js'>CardHeader</abbr> | `16.879 kB`<br />`6.664 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardPreview.fixture.js'>CardPreview</abbr> | `14.42 kB`<br />`5.934 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr> | `69.141 kB`<br />`20.157 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `217.326 kB`<br />`63.005 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr> | `44.442 kB`<br />`14.607 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr> | `8.39 kB`<br />`2.64 kB` |
| <samp>react-timepicker-compat</samp> <br /> <abbr title='bundle-size/TimePicker.fixture.js'>TimePicker</abbr> | `106.751 kB`<br />`35.596 kB` |
</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/aa652de894084d5e10cbc708aa4542cc46916856'>aa652de894084d5e10cbc708aa4542cc46916856</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32200/perf-test-react-components/Avatar-mount_base.html">648</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32200/perf-test-react-components/Avatar-mount.html">637</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32200/perf-test-react-components/Button-mount_base.html">297</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32200/perf-test-react-components/Button-mount.html">314</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32200/perf-test-react-components/Field-mount_base.html">1147</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32200/perf-test-react-components/Field-mount.html">1127</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32200/perf-test-react-components/FluentProvider-mount_base.html">723</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32200/perf-test-react-components/FluentProvider-mount.html">704</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32200/perf-test-react-components/FluentProviderWithTheme-mount_base.html">83</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32200/perf-test-react-components/FluentProviderWithTheme-mount.html">88</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32200/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">36</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32200/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">38</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32200/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">86</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32200/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">78</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32200/perf-test-react-components/MakeStyles-mount_base.html">878</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32200/perf-test-react-components/MakeStyles-mount.html">857</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32200/perf-test-react-components/Persona-mount_base.html">1767</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32200/perf-test-react-components/Persona-mount.html">1765</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32200/perf-test-react-components/SpinButton-mount_base.html">1379</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32200/perf-test-react-components/SpinButton-mount.html">1410</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32200/perf-test-react-components/SwatchPicker-mount_base.html">1680</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32200/perf-test-react-components/SwatchPicker-mount.html">1672</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

File: change/@fluentui-react-card-83adc37b-ffe3-4cbd-a1f8-93b6f41549a3.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (fabricteam) [inline change/@fluentui-react-card-83adc37b-ffe3-4cbd-a1f8-93b6f41549a3.json:1]: ## 🕵🏾‍♀️ [<b data-uid='vrt-gist'></b> visual regressions to review in the fluentuiv9 Visual Regression Report](https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=32200&commitId=db00900378b54e9d41aa456c95b192c59c5c66a5&env=prod&clientType=FLUENTUI)


  <details><summary><b>Avatar Converged</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Avatar Converged.badgeMask - RTL.normal.chromium.png|2| Changed|

</details>


  <details><summary><b>Card Converged - Interactive</b> 3 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Card Converged - Interactive.appearance interactive - Outline - Dark Mode.click.chromium.png|0| Removed|
|Card Converged - Interactive.appearance interactive - Outline - Dark Mode.hover.chromium.png|0| Removed|
|Card Converged - Interactive.appearance interactive - Outline - Dark Mode.normal.chromium.png|0| Removed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

**Review by marcosmoura — APPROVED** [review]

File: change/@fluentui-react-card-83adc37b-ffe3-4cbd-a1f8-93b6f41549a3.json (hunk 1 above)
Comment (smhigley) [reply change/@fluentui-react-card-83adc37b-ffe3-4cbd-a1f8-93b6f41549a3.json:1]: not related to PR

**Pull request merged**